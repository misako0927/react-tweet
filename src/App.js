import React, { Component } from 'react'
import './App.css'
import Header from './Header'
import Footer from './Footer'
import Contents from './Contents'

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <Contents tweets={this.props.tweets} />
        <Footer />
      </div>
    )
  }
}

export default App
