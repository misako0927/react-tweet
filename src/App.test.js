import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import renderer from 'react-test-renderer'
import { createRenderer } from 'react-test-renderer/shallow'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
})
it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
it('renders with shallow', () => {
  const shallow = createRenderer()
  shallow.render(<App />)
  const code = <code>src/App.js</code>
  expect(shallow.getRenderOutput().type).toEqual('div')
})