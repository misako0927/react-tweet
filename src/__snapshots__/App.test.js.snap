exports[`test renders correctly 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro">
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <div>
    <div
      className="tweet-contents" />
    <div>
      <button
        onClick={[Function]}>
        Please Click
      </button>
    </div>
  </div>
  <div
    className="App-footer">
    <p>
      this is footer
    </p>
  </div>
</div>
`;
